File: Array-String/HashTable/CPP/find_resultant_array_after_removing_anagrams.cpp
#include <algorithm>
#include <vector>
#include <string>
using namespace std;

class Solution {
    public:
        vector<string> removeAnagrams(vector<string>& words) {
            int n = words.size();
            vector<string> res;
            string prev;

            for (string s: words) {
                string cur = s;
                sort(cur.begin(), cur.end());
                if (prev != cur) {
                    res.push_back(s);
                    prev = cur;
                }
            }
            return res;
        }
    };

File: Array-String/HashTable/Python/find_resultant_array_after_removing_anagrams.py
class Solution:
    def removeAnagrams(self, words: list[str]) -> list[str]:
        n, res = len(words), []
        prev = None
        for w in words:
            cur = sorted(w)
            if prev != cur:
                res.append(w)
                prev = cur
        return res